sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/KalixServer.scala

```diff
@@ -403,7 +405,29 @@ case class KalixServer(applicationContext: ApplicationContext, config: Config) {
 
         setKalixClient(clz, webClientProviderHolder)
 
-        kalixBeanFactory.getBean(clz)
+        val workflowEntity = kalixBeanFactory.getBean(clz)
+
+        val workflowStateType: Class[S] =
+          workflowEntity.getClass.getGenericSuperclass
+            .asInstanceOf[ParameterizedType]
+            .getActualTypeArguments
+            .head
```

Comment:
Could it be the case that no `getActualTypeArguments` is empty?

Reply:
no, since we managed that class

Reply:
I see. You mean it has to be a `public static class Workflow<S> ` and therefore it has one parameter.

Reply:
correct